packages/babel-core/src/transformation/file/options/option-manager.js

```diff
@@ -174,6 +174,10 @@ export default class OptionManager {
     loc = loc || alias;
 
     for (let key in opts) {
+      // Ignore Babel's ES2015 module sentinel on the `opts` object, if any. This
+      // is so Babel presets may be written using ES2015 modules syntax.
+      if (key === "__esModule") continue;
```

Comment:
Hmm, red herring. But the time we get here, if (val.__esModel) then val.default === undefined.

I tried quite a few things prior to my submission and the only thing that worked and passed all tests was what I submitted. The code is somewhat brittle and it requires a delicate tweak to achieve what you are more in favor of.  That being said I do concur with your aspiration so I'll keep at it for a while longer (though already it has taken way too much time for such a tiny task).